diff -urNp -x '*.orig' zabbix-5.2.3.org/configure.ac zabbix-5.2.3/configure.ac
--- zabbix-5.2.3.org/configure.ac	2020-12-21 10:47:57.000000000 +0100
+++ zabbix-5.2.3/configure.ac	2021-08-24 11:18:54.266877449 +0200
@@ -1672,7 +1672,7 @@ AM_CONDITIONAL(HAVE_IPMI, [test "x$have_
 AM_CONDITIONAL(HAVE_SSH2, [test "x$have_ssh" = "xyes (libssh2)"])
 
 dnl Check if Zabbix internal IPC services are used
-have_ipcservice="no"
+have_ipcservice="yes"
 if test "x$have_ipmi" = "xyes"; then
 	have_ipcservice="yes"
 fi
