diff -urNp -x '*.orig' zabbix-5.2.3.org/conf/zabbix_agentd.conf zabbix-5.2.3/conf/zabbix_agentd.conf
--- zabbix-5.2.3.org/conf/zabbix_agentd.conf	2020-12-21 10:47:57.000000000 +0100
+++ zabbix-5.2.3/conf/zabbix_agentd.conf	2021-08-24 11:18:50.376675510 +0200
@@ -9,6 +9,7 @@
 # Mandatory: no
 # Default:
 # PidFile=/tmp/zabbix_agentd.pid
+PidFile=/var/run/zabbix/zabbix_agentd.pid
 
 ### Option: LogType
 #	Specifies where log messages are written to:
@@ -27,7 +28,7 @@
 # Default:
 # LogFile=
 
-LogFile=/tmp/zabbix_agentd.log
+LogFile=/var/log/zabbix/zabbix_agentd.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -304,7 +305,7 @@ Hostname=Zabbix server
 
 # Include=/usr/local/etc/zabbix_agentd.userparams.conf
 # Include=/usr/local/etc/zabbix_agentd.conf.d/
-# Include=/usr/local/etc/zabbix_agentd.conf.d/*.conf
+Include=/etc/zabbix/zabbix_agentd.conf.d/*.conf
 
 ####### USER-DEFINED MONITORED PARAMETERS #######
 
diff -urNp -x '*.orig' zabbix-5.2.3.org/conf/zabbix_proxy.conf zabbix-5.2.3/conf/zabbix_proxy.conf
--- zabbix-5.2.3.org/conf/zabbix_proxy.conf	2020-12-21 10:47:57.000000000 +0100
+++ zabbix-5.2.3/conf/zabbix_proxy.conf	2021-08-24 11:18:50.376675510 +0200
@@ -88,7 +88,7 @@ Hostname=Zabbix proxy
 # Default:
 # LogFile=
 
-LogFile=/tmp/zabbix_proxy.log
+LogFile=/var/log/zabbix/zabbix_proxy.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -136,7 +136,7 @@ LogFile=/tmp/zabbix_proxy.log
 #
 # Mandatory: no
 # Default:
-# PidFile=/tmp/zabbix_proxy.pid
+PidFile=/var/run/zabbix/zabbix_proxy.pid
 
 ### Option: SocketDir
 #	IPC socket directory.
@@ -586,7 +586,7 @@ LogSlowQueries=3000
 
 # Include=/usr/local/etc/zabbix_proxy.general.conf
 # Include=/usr/local/etc/zabbix_proxy.conf.d/
-# Include=/usr/local/etc/zabbix_proxy.conf.d/*.conf
+Include=/etc/zabbix/zabbix_proxy.conf.d/*.conf
 
 ### Option: SSLCertLocation
 #	Location of SSL client certificates.
diff -urNp -x '*.orig' zabbix-5.2.3.org/conf/zabbix_server.conf zabbix-5.2.3/conf/zabbix_server.conf
--- zabbix-5.2.3.org/conf/zabbix_server.conf	2020-12-21 10:47:57.000000000 +0100
+++ zabbix-5.2.3/conf/zabbix_server.conf	2021-08-24 11:18:50.376675510 +0200
@@ -35,7 +35,7 @@
 # Default:
 # LogFile=
 
-LogFile=/tmp/zabbix_server.log
+LogFile=/var/log/zabbix/zabbix_server.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -65,7 +65,7 @@ LogFile=/tmp/zabbix_server.log
 #
 # Mandatory: no
 # Default:
-# PidFile=/tmp/zabbix_server.pid
+PidFile=/var/run/zabbix/zabbix_server.pid
 
 ### Option: SocketDir
 #	IPC socket directory.
@@ -620,7 +620,7 @@ LogSlowQueries=3000
 
 # Include=/usr/local/etc/zabbix_server.general.conf
 # Include=/usr/local/etc/zabbix_server.conf.d/
-# Include=/usr/local/etc/zabbix_server.conf.d/*.conf
+Include=/etc/zabbix/zabbix_server.conf.d/*.conf
 
 ### Option: SSLCertLocation
 #	Location of SSL client certificates.
diff -urNp -x '*.orig' zabbix-5.2.3.org/src/zabbix_java/lib/logback.xml zabbix-5.2.3/src/zabbix_java/lib/logback.xml
--- zabbix-5.2.3.org/src/zabbix_java/lib/logback.xml	2020-12-14 11:27:17.000000000 +0100
+++ zabbix-5.2.3/src/zabbix_java/lib/logback.xml	2021-08-24 11:18:50.376675510 +0200
@@ -2,10 +2,10 @@
 
 	<appender name="FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
 
-		<file>/tmp/zabbix_java.log</file>
+		<file>/var/log/zabbix/zabbix_java.log</file>
 
 		<rollingPolicy class="ch.qos.logback.core.rolling.FixedWindowRollingPolicy">
-			<fileNamePattern>/tmp/zabbix_java.log.%i</fileNamePattern>
+			<fileNamePattern>/var/log/zabbix/zabbix_java.log.%i</fileNamePattern>
 			<minIndex>1</minIndex>
 			<maxIndex>3</maxIndex>
 		</rollingPolicy>
diff -urNp -x '*.orig' zabbix-5.2.3.org/src/zabbix_java/settings.sh zabbix-5.2.3/src/zabbix_java/settings.sh
--- zabbix-5.2.3.org/src/zabbix_java/settings.sh	2020-12-14 11:27:17.000000000 +0100
+++ zabbix-5.2.3/src/zabbix_java/settings.sh	2021-08-24 11:18:50.376675510 +0200
@@ -24,7 +24,7 @@
 # Default:
 # PID_FILE=
 
-PID_FILE="/tmp/zabbix_java.pid"
+PID_FILE="/var/run/zabbix/zabbix_java.pid"
 
 ### Option: zabbix.startPollers
 #	Number of worker threads to start.
diff -urNp -x '*.orig' zabbix-5.2.3.org/ui/include/classes/setup/CSetupWizard.php zabbix-5.2.3/ui/include/classes/setup/CSetupWizard.php
--- zabbix-5.2.3.org/ui/include/classes/setup/CSetupWizard.php	2020-12-21 10:47:57.000000000 +0100
+++ zabbix-5.2.3/ui/include/classes/setup/CSetupWizard.php	2021-08-24 11:18:50.376675510 +0200
@@ -604,7 +604,7 @@ class CSetupWizard extends CForm {
 
 		$this->setConfig('ZBX_CONFIG_FILE_CORRECT', true);
 
-		$config_file_name = APP::getRootDir().CConfigFile::CONFIG_FILE_PATH;
+		$config_file_name = '/etc/zabbix/frontend/zabbix.conf.php';
 		$config = new CConfigFile($config_file_name);
 		$config->config = [
 			'DB' => [
