--- zabbix-3.2.0.orig/conf/zabbix_agentd.conf	2016-09-13 14:20:55.000000000 +0200
+++ zabbix-3.2.0/conf/zabbix_agentd.conf	2016-10-01 18:42:28.000000000 +0200
@@ -9,6 +9,7 @@
 # Mandatory: no
 # Default:
 # PidFile=/tmp/zabbix_agentd.pid
+PidFile=/var/run/zabbix/zabbix_agentd.pid
 
 ### Option: LogType
 #	Specifies where log messages are written to:
@@ -27,7 +28,7 @@
 # Default:
 # LogFile=
 
-LogFile=/tmp/zabbix_agentd.log
+LogFile=/var/log/zabbix/zabbix_agentd.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -260,7 +261,7 @@
 
 # Include=/usr/local/etc/zabbix_agentd.userparams.conf
 # Include=/usr/local/etc/zabbix_agentd.conf.d/
-# Include=/usr/local/etc/zabbix_agentd.conf.d/*.conf
+Include=/etc/zabbix/zabbix_agentd.conf.d/*.conf
 
 ####### USER-DEFINED MONITORED PARAMETERS #######
 
--- zabbix-3.2.0.orig/conf/zabbix_proxy.conf	2016-09-13 14:20:55.000000000 +0200
+++ zabbix-3.2.0/conf/zabbix_proxy.conf	2016-10-01 18:42:44.000000000 +0200
@@ -82,7 +82,7 @@
 # Default:
 # LogFile=
 
-LogFile=/tmp/zabbix_proxy.log
+LogFile=/var/log/zabbix/zabbix_proxy.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -112,7 +112,7 @@
 #
 # Mandatory: no
 # Default:
-# PidFile=/tmp/zabbix_proxy.pid
+PidFile=/var/run/zabbix/zabbix_proxy.pid
 
 ### Option: DBHost
 #	Database host name.
@@ -543,7 +543,7 @@
 
 # Include=/usr/local/etc/zabbix_proxy.general.conf
 # Include=/usr/local/etc/zabbix_proxy.conf.d/
-# Include=/usr/local/etc/zabbix_proxy.conf.d/*.conf
+Include=/etc/zabbix/zabbix_proxy.conf.d/*.conf
 
 ### Option: SSLCertLocation
 #	Location of SSL client certificates.
--- zabbix-3.2.0.orig/conf/zabbix_server.conf	2016-09-13 14:20:55.000000000 +0200
+++ zabbix-3.2.0/conf/zabbix_server.conf	2016-10-01 18:43:07.000000000 +0200
@@ -35,7 +35,7 @@
 # Default:
 # LogFile=
 
-LogFile=/tmp/zabbix_server.log
+LogFile=/var/log/zabbix/zabbix_server.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -65,7 +65,7 @@
 #
 # Mandatory: no
 # Default:
-# PidFile=/tmp/zabbix_server.pid
+PidFile=/var/run/zabbix/zabbix_server.pid
 
 ### Option: DBHost
 #	Database host name.
@@ -552,7 +552,7 @@
 
 # Include=/usr/local/etc/zabbix_server.general.conf
 # Include=/usr/local/etc/zabbix_server.conf.d/
-# Include=/usr/local/etc/zabbix_server.conf.d/*.conf
+Include=/etc/zabbix/zabbix_server.conf.d/*.conf
 
 ### Option: SSLCertLocation
 #	Location of SSL client certificates.
--- zabbix-3.2.0.orig/frontends/php/include/classes/setup/CSetupWizard.php	2016-09-13 14:21:05.000000000 +0200
+++ zabbix-3.2.0/frontends/php/include/classes/setup/CSetupWizard.php	2016-10-01 18:42:06.000000000 +0200
@@ -386,7 +386,7 @@
 				new CTag('p', true, _('Alternatively, you can install it manually:')),
 				new CTag('ol', true, [
 					new CTag('li', true, new CLink(_('Download the configuration file'), 'setup.php?save_config=1')),
-					new CTag('li', true, _s('Save it as "%1$s"', $config_file_name))
+					new CTag('li', true, _s('Save it as "%1$s"', '/etc/zabbix/frontend/zabbix.conf.php'))
 				]),
 			];
 		}
--- zabbix-3.2.0.orig/src/zabbix_java/lib/logback.xml	2016-09-13 14:20:55.000000000 +0200
+++ zabbix-3.2.0/src/zabbix_java/lib/logback.xml	2016-10-01 18:42:06.000000000 +0200
@@ -2,10 +2,10 @@
 
 	<appender name="FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
 
-		<file>/tmp/zabbix_java.log</file>
+		<file>/var/log/zabbix/zabbix_java.log</file>
 
 		<rollingPolicy class="ch.qos.logback.core.rolling.FixedWindowRollingPolicy">
-			<fileNamePattern>/tmp/zabbix_java.log.%i</fileNamePattern>
+			<fileNamePattern>/var/log/zabbix/zabbix_java.log.%i</fileNamePattern>
 			<minIndex>1</minIndex>
 			<maxIndex>3</maxIndex>
 		</rollingPolicy>
--- zabbix-3.2.0.orig/src/zabbix_java/settings.sh	2016-09-13 14:20:55.000000000 +0200
+++ zabbix-3.2.0/src/zabbix_java/settings.sh	2016-10-01 18:42:06.000000000 +0200
@@ -24,7 +24,7 @@
 # Default:
 # PID_FILE=
 
-PID_FILE="/tmp/zabbix_java.pid"
+PID_FILE="/var/run/zabbix/zabbix_java.pid"
 
 ### Option: zabbix.startPollers
 #	Number of worker threads to start.
