--- zabbix-5.2.3/src/go/Makefile.am~	2020-12-14 12:27:17.000000000 +0200
+++ zabbix-5.2.3/src/go/Makefile.am	2021-08-24 15:25:21.065256983 +0300
@@ -26,7 +26,7 @@
 	rm -f bin/zabbix_agent2 bin/mock_server
 
 install:
-	CGO_CFLAGS="${CGO_CFLAGS}" CGO_LDFLAGS="${CGO_LDFLAGS}" GOBIN=${GOBIN} \
+	CGO_CFLAGS="${CGO_CFLAGS}" CGO_LDFLAGS="${CGO_LDFLAGS}" GOBIN=$(DESTDIR)${GOBIN} \
 		go install -ldflags="${GOLDFLAGS}" zabbix.com/cmd/zabbix_agent2
 	test -f "$(DESTDIR)@AGENT2_CONFIG_FILE@" || cp "conf/zabbix_agent2.conf" "$(DESTDIR)@AGENT2_CONFIG_FILE@"
 
