--- zabbix-7.0.3/src/go/Makefile.am.orig	2024-08-22 15:13:38.617557743 +0200
+++ zabbix-7.0.3/src/go/Makefile.am	2024-08-22 15:16:13.296462739 +0200
@@ -50,6 +50,7 @@
 bin/zabbix_agent2: FORCE
 	CGO_CFLAGS="${CGO_CFLAGS}" CGO_LDFLAGS="${CGO_LDFLAGS}" \
 		$(GO) build \
+		-mod=vendor \
 		-tags="${AGENT_GOTAGS}" \
 		-ldflags="${AGENT_GOLDFLAGS}" \
 		-o bin \
@@ -57,6 +58,7 @@
 
 bin/zabbix_web_service: FORCE
 	$(GO) build \
+	-mod=vendor \
 	-ldflags="${WEBSERVICE_GOLDFLAGS}" \
 	-o bin \
 	golang.zabbix.com/agent2/cmd/zabbix_web_service
