diff -durN zabbix-3.4.4.orig/src/zabbix_server/Makefile.am zabbix-3.4.4/src/zabbix_server/Makefile.am
--- zabbix-3.4.4.orig/src/zabbix_server/Makefile.am	2017-11-09 11:37:30.000000000 +0100
+++ zabbix-3.4.4/src/zabbix_server/Makefile.am	2017-12-14 13:26:08.000000000 +0100
@@ -51,7 +51,6 @@
 	selfmon/libzbxselfmon.a \
 	vmware/libzbxvmware.a \
 	taskmanager/libzbxtaskmanager.a \
-	ipmi/libipmi.a \
 	scripts/libzbxscripts.a \
 	preprocessor/libpreprocessor.a \
 	$(top_srcdir)/src/libs/zbxsysinfo/libzbxserversysinfo.a \
@@ -75,6 +74,7 @@
 	$(top_srcdir)/src/libs/zbxjson/libzbxjson.a \
 	$(top_srcdir)/src/libs/zbxhttp/libzbxhttp.a \
 	$(top_srcdir)/src/libs/zbxipcservice/libzbxipcservice.a \
+	ipmi/libipmi.a \
 	$(top_srcdir)/src/libs/zbxexec/libzbxexec.a \
 	$(top_srcdir)/src/libs/zbxicmpping/libzbxicmpping.a \
 	$(top_srcdir)/src/libs/zbxdbupgrade/libzbxdbupgrade.a \
