diff -dur -x '*~' -x '*.orig' -x '*.rej' zabbix-3.4.4.orig/conf/zabbix_proxy.conf zabbix-3.4.4/conf/zabbix_proxy.conf
--- zabbix-3.4.4.orig/conf/zabbix_proxy.conf	2017-12-14 13:15:25.000000000 +0100
+++ zabbix-3.4.4/conf/zabbix_proxy.conf	2017-12-14 13:15:42.000000000 +0100
@@ -152,6 +152,7 @@
 ### Option: DBName
 #	Database name.
 #	For SQLite3 path to database file must be provided. DBUser and DBPassword are ignored.
+#	If provided name does not contain '/' or ':' (no a path or special type) a .db file in /var/lib/zabbix will be used.
 #	Warning: do not attempt to use the same database Zabbix server is using.
 #
 # Mandatory: yes
diff -dur -x '*~' -x '*.orig' -x '*.rej' zabbix-3.4.4.orig/src/libs/zbxdb/db.c zabbix-3.4.4/src/libs/zbxdb/db.c
--- zabbix-3.4.4.orig/src/libs/zbxdb/db.c	2017-11-09 11:37:31.000000000 +0100
+++ zabbix-3.4.4/src/libs/zbxdb/db.c	2017-12-14 13:16:29.000000000 +0100
@@ -294,6 +294,7 @@
 	DB_ROW		row;
 #elif defined(HAVE_SQLITE3)
 	char		*p, *path = NULL;
+	char		dbname_buf[1024];
 #endif
 
 #ifndef HAVE_MYSQL
@@ -831,6 +831,10 @@ out:
 	ZBX_UNUSED(ca);
 	ZBX_UNUSED(cipher);
 	ZBX_UNUSED(cipher_13);
+	if (!strchr(dbname, '/') && !strchr(dbname, ':')) {
+		zbx_snprintf(dbname_buf, sizeof(dbname_buf), "/var/lib/zabbix/%s.db", dbname);
+		dbname = dbname_buf;
+	}
 #ifdef HAVE_FUNCTION_SQLITE3_OPEN_V2
 	if (SQLITE_OK != sqlite3_open_v2(dbname, &conn, SQLITE_OPEN_READWRITE, NULL))
 #else

