--- zabbix-5.4.3/conf/zabbix_proxy.conf~	2021-08-24 12:51:15.000000000 +0300
+++ zabbix-5.4.3/conf/zabbix_proxy.conf	2021-08-24 12:52:01.573388850 +0300
@@ -160,6 +160,7 @@
 ### Option: DBName
 #	Database name.
 #	For SQLite3 path to database file must be provided. DBUser and DBPassword are ignored.
+#	If provided name does not contain '/' or ':' (no a path or special type) a .db file in /var/lib/zabbix will be used.
 #	If the Net Service Name connection method is used to connect to Oracle database, specify the service name from
 #	the tnsnames.ora file or set to empty string; also see the TWO_TASK environment variable if DBName is set to
 #	empty string.
diff -urNp -x '*.orig' zabbix-5.2.3.org/src/libs/zbxdb/db.c zabbix-5.2.3/src/libs/zbxdb/db.c
--- zabbix-5.2.3.org/src/libs/zbxdb/db.c	2020-12-14 11:27:17.000000000 +0100
+++ zabbix-5.2.3/src/libs/zbxdb/db.c	2021-08-24 11:18:52.356778296 +0200
@@ -383,6 +383,7 @@ int	zbx_db_connect(char *host, char *use
 	unsigned int	i = 0;
 #elif defined(HAVE_SQLITE3)
 	char		*p, *path = NULL;
+	char		dbname_buf[1024];
 #endif
 
 #ifndef HAVE_MYSQL
@@ -830,6 +831,10 @@ out:
 	ZBX_UNUSED(cipher);
 	ZBX_UNUSED(cipher_13);
 	ZBX_UNUSED(read_only_recoverable);
+	if (!strchr(dbname, '/') && !strchr(dbname, ':')) {
+		zbx_snprintf(dbname_buf, sizeof(dbname_buf), "/var/lib/zabbix/%s.db", dbname);
+		dbname = dbname_buf;
+	}
 #ifdef HAVE_FUNCTION_SQLITE3_OPEN_V2
 	if (SQLITE_OK != sqlite3_open_v2(dbname, &conn, SQLITE_OPEN_READWRITE, NULL))
 #else
