--- zabbix-7.0.20/ui/widgets/svggraph/includes/CSvgGraphHelper.php.orig	2025-10-28 08:32:37.785793008 +0100
+++ zabbix-7.0.20/ui/widgets/svggraph/includes/CSvgGraphHelper.php	2025-10-30 20:24:57.907138212 +0100
@@ -154,7 +154,7 @@
 			$resolve_macros = $templateid === '' || $override_hostid !== '';
 
 			$options = [
-				'output' => ['itemid', 'hostid', 'type', 'master_itemid', 'delay', 'history', 'trends', 'units',
+				'output' => ['itemid', 'hostid', 'type', 'key_', 'master_itemid', 'delay', 'history', 'trends', 'units',
 					'value_type'
 				],
 				'selectPreprocessing' => ['type', 'params'],
@@ -299,7 +299,7 @@
 			$resolve_macros = $templateid === '' || $override_hostid !== '';
 
 			$db_items = API::Item()->get([
-				'output' => ['itemid', 'hostid', 'type', 'master_itemid', 'delay',
+				'output' => ['itemid', 'hostid', 'type', 'key_', 'master_itemid', 'delay',
 					$resolve_macros ? 'name_resolved' : 'name', 'history', 'trends', 'units', 'value_type'
 				],
 				'selectPreprocessing' => ['type', 'params'],
